### PR TITLE
[failproofai-626] Support multiple custom policy files

### PR DESCRIPTION
**Pull request by NiveditJain**
## Summary
- add ordered, additive `customPoliciesPaths` configuration while retaining legacy `customPoliciesPath` support
- allow repeated `--custom` flags and load, list, remove, and display all configured files
- add independent dashboard toggles for every policy in explicit and convention files
- persist off policies as source-qualified IDs, so duplicate names across files/scopes remain independently controllable
- exclude off policies before event matching and function execution
- update dashboard, documentation, changelog, unit tests, and end-to-end coverage

## Toggle edge-case coverage
- multiple explicit files, including duplicate policy names
- duplicate convention names at project and user scope
- sibling policies in the same file remain active
- disabled IDs merged across project/local/global scopes
- legacy singular and current plural custom-path config
- stale disabled IDs are harmless
- re-enabling removes the persisted opt-out

## Validation
- TypeScript typecheck passed
- ESLint passed with 5 pre-existing warnings and 0 errors
- focused unit/integration suites: 105 passed
- custom-policy end-to-end suite: 51 passed
- repository-wide unit run: 2460 passed; 18 pre-existing HOME/filesystem-sensitive failures reproduced in unrelated cache, Hermes, convention-sync, and first-run suites
- CI and supply-chain checks passed

## Summary by CodeRabbit

* **New Features**
  * Added repeating `--custom` support for multiple explicit custom policy files, with correct install/uninstall and load order.
  * Dashboard and runtime now apply independent enable/disable toggles per explicit/convention policy file, backed by merged `disabledCustomPolicies`.
  * Configuration now supports `customPoliciesPaths` (array) with legacy `customPoliciesPath` retained.
* **Documentation**
  * Updated CLI help and custom policy/configuration docs for multi-file and toggle behavior.
* **Bug Fixes**
  * Fixed dashboard toggle labeling/state display, corrected the detail-panel disclosure glyph, and prevented Configure tab toggle state from getting stuck.
* **Tests**
  * Expanded E2E and unit coverage for multi-path loading, deduping, and disabled policy merge/toggle rules.

## Latest CLI status fix
- `failproofai policies` now renders dashboard-disabled explicit policies as `OFF` without a green checkmark
- convention files render `OFF` when all policies are disabled and `MIXED` when enabled and disabled siblings coexist; disabled names are labeled `(OFF)`
- clean Git-equipped Docker E2E: 314 passed, 6 intentionally skipped
- clean Docker production build, npm package, global install, and two-file custom-policy discovery passed
- fix the Policies dashboard event-detail disclosure so it renders the intended ▾ glyph instead of literal Unicode escape text
- merged-scope dashboard correctness: displayed state matches runtime, enabling clears contributing project/local/user opt-outs, and unreadable explicit files are isolated
- fix the Policies loading ellipsis so it renders … instead of literal Unicode escape text

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 0.0.15-beta.1 — 2026-07-29
+
+### Features
+- Support multiple explicit custom policy files. Repeat `--custom <path>` during installation to configure an ordered list stored as `customPoliciesPaths`; runtime loading, CLI output, uninstall, and the dashboard now handle every file. Existing `customPoliciesPath` configurations remain supported.
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply CHANGELOG.md]: Fixed in 039c7d7: the changelog entry now includes (#623).

File: docs/custom-policies.mdx [hunk 2]
```diff
@@ -72,17 +72,20 @@ failproofai policies --install --custom ./my-policies.js
 # Replace the policies file path
 failproofai policies --install --custom ./new-policies.js
 
+# Configure multiple explicit files (loaded in flag order)
+failproofai policies --install --custom ./security.js --custom ./workflow.js
+
 # Remove the custom policies path from config
 failproofai policies --uninstall --custom
 ```
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply docs/custom-policies.mdx:80]: Fixed in 039c7d7: the explicit-path documentation now consistently uses plural paths/files terminology.

File: src/hooks/custom-hooks-loader.ts [hunk 3]
```diff
@@ -228,16 +228,21 @@ export async function loadAllCustomHooks(
...
+    : customPoliciesPaths ?? []) {
     const absPath = isAbsolute(customPoliciesPath)
       ? customPoliciesPath
       : resolve(projectRoot, customPoliciesPath);
     if (existsSync(absPath)) {
-      loadedPaths.add(absPath);
-      await loadSingleFile(absPath);
+      if (!loadedPaths.has(absPath)) {
+        loadedPaths.add(absPath);
+        await loadSingleFile(absPath);
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply src/hooks/custom-hooks-loader.ts:251]: Verified against the runtime architecture: each hook event is handled by a new short-lived  process, so Bun's in-process ESM cache cannot persist between events.  and  only reset registries within that one invocation before its first imports. The  query is needed for repeated loads inside one process (notably CLI listing/tests), and the docs' per-event fresh-load statement remains accurate. No code change is appropriate for this finding.

[the remark this answers is not in the document]
Comment (NiveditJain) [reply src/hooks/custom-hooks-loader.ts:251]: Fixed in 039c7d7: every explicit path now goes through , including absolute inputs, and a regression test covers an absolute  alias overlapping a convention file.

File: src/hooks/manager.ts [hunk 4]
```diff
@@ -212,50 +212,57 @@ async function installHooksImpl(
...
+        `\nValidated ${validatedHooks.length} custom hook(s) from ${path}: ${validatedHooks.map((h) => h.name).join(", ")}`,
       );
-      process.exit(1);
     }
-    console.log(
-      `\nValidated ${validatedHooks.length} custom hook(s): ${validatedHooks.map((h) => h.name).join(", ")}`,
-    );
   }
   writeScopedHooksConfig(configToWrite, scope, cwd);
   console.log(`\nEnabled ${selectedPolicies.length} policy(ies): ${selectedPolicies.join(", ")}\n`);
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply src/hooks/manager.ts:291]: Fixed in 039c7d7: an empty path array is now treated as absent, preserving legacy/lower-scope configuration and skipping the validation/write branch; covered by a unit test.

File: app/actions/get-hooks-config.ts [hunk 5]
```diff
@@ -166,6 +177,7 @@ function buildEventScope(match: { events?: string[]; toolNames?: string[] }): st
 export async function getHooksConfigAction(): Promise<HooksConfigPayload> {
   const config = readHooksConfig();
   const enabledSet = new Set(config.enabledPolicies);
+  const disabledCustomPolicies = new Set(config.disabledCustomPolicies ?? []);
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply app/actions/get-hooks-config.ts:183]: Fixed in 8d0c306. The dashboard now reads readMergedHooksConfig using the launch cwd, and enabling a custom policy removes its disabled ID from every project/local/user scope that contributes it. Added regression coverage for multi-scope removal.

File: app/actions/get-hooks-config.ts [hunk 6]
```diff
@@ -195,20 +207,31 @@ export async function getHooksConfigAction(): Promise<HooksConfigPayload> {
...
+  const customPoliciesPaths = config.customPoliciesPaths ?? (config.customPoliciesPath ? [config.customPoliciesPath] : []);
+  const launchRoot = findProjectConfigDir(process.env.FAILPROOFAI_LAUNCH_CWD || process.cwd());
+  const resolvedCustomPaths = customPoliciesPaths.map((path) => resolve(launchRoot, path));
+  const parsedCustomPolicies = await Promise.all(resolvedCustomPaths.map(async (path) =>
+    (await parseCustomPoliciesFromFile(path)).map((policy) => {
+      const id = customPolicyId(path, policy.name);
+      return { ...policy, id, enabled: !disabledCustomPolicies.has(id) };
+    })
+  ));
+  const customPolicies = parsedCustomPolicies.flat();
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply app/actions/get-hooks-config.ts]: Fixed in 8d0c306. Explicit policy parsing now catches failures per path and returns an empty list for only that file, preserving all successfully parsed files and keeping the dashboard payload available.

**Review by nk-ag — DISMISSED** [review]

**Review by nk-ag — APPROVED** [review]
